### ISSUE TITLE
Search function does not work with rails alias_attribute

### ISSUE THREAD

**Issue by JavierM42** [thoughtbot/administrate#1692]
I used `alias_attribute` to change an attribute name at the model level (my db is shared so I can't simply rename the column). I then changed the `attribute_types`, `collection_attributes` and `show_page_attributes` to the new name, and everything worked but the search, which throws

```
PG::UndefinedColumn: ERROR:  column table.aliased_column does not exist
LINE 1: ... AS CHAR(256))) LIKE '%search_query%' OR LOWER(CAST("...
```

  - Rails 6.0.3.2
  - administrate 0.13.0

I know that I can change labels to make the user-facing text be whatever I want, but it would be nice to support aliases to improve the developer experience.

**Comment by nickcharlton** [comment]
Yeah, this is something we've not implemented yet but would be good to do.

Is it just search where you see this fail, or are there other problems with `alias_attribute`?

**Comment by JavierM42** [comment]
We've noticed problems only with `alias_attribute` so far.

**Comment by nickcharlton** [comment]
Oh, no, I mean: Did you see any other problems apart from this one relating to search with `alias_attribute`?

**Comment by JavierM42** [comment]
My bad, I understood the question correctly and typed "alias_attribute" instead of "the search".

We have not found any problems but the search while using alias attribute.

**Comment by nickcharlton** [comment]
Great, thanks!

**Comment by bekicot** [comment]
For a "quick solution" I usually add `searchable: false` to the aliased attribute and `searchable: true` to the original attribute.

**Comment by januszm** [comment]
Would it be possible to move this to the presentation layer and e.g. add an option to the alias field in `with_options` ?

E.g. when there's a column named `myother_id`, Administrate will consider it as reference to another model and display it as `Myother` in the view. I'd like to show it as `MyotherID`, this could potentially be passed to with_options:

```
myother_id: Field::String.with_options(as: "MyOtherId"), # or alias: or label:
```
